src/components/CharacterWithQuote/CharacterWithQuote.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import { getQuote } from '../../utils/quotesService';
+import extractQuoteIDFromPath from '../../utils/extractQuoteIDFromPath';
+import extractActionFromPath from '../../utils/extractActionFromPath';
+import extractRoleFromPath from '../../utils/extractRoleFromPath';
+import { history } from '../../types';
+
+function CharacterWitQuote({ history, children }) {
+  const role = extractRoleFromPath(history.location.pathname);
+  const pose = extractActionFromPath(history.location.pathname);
+  const quoteID = extractQuoteIDFromPath(history.location.pathname);
+  const quote = getQuote(role, pose, quoteID);
+
+  function renderChildren(child, index) {
+    if (index === 0) {
+      return React.cloneElement(children[index], { quote });
+    } else if (index === 1) return React.cloneElement(children[index], { pose });
+  }
+
+  return <>{React.Children.map(children, renderChildren)}</>;
```

Comment:
I think it is not required to wrap it with fragment.

Reply:
True! Thanks! 👍 
https://github.com/BuildIt-Poland/daily-routine/pull/75/commits/3a535146c57bfef269a3f85235bc413009812cad#diff-d1a8069f8bac699dcb367f774d57206a